private/getActualTimings.m

```diff
@@ -1,21 +1,61 @@
 function [durationFirstPartActual, durationZeroGradientActual, durationSecondPartActual, totalTimeActual, zeroGradientAtIndex] = ...
-    getActualTimings(durationFirstPartRequested, durationZeroGradientRequested, durationSecondPartRequested, discretizationSteps)
-totalTime = durationFirstPartRequested + durationSecondPartRequested + durationZeroGradientRequested;
-dt = totalTime/discretizationSteps;
+    getActualTimings(durationFirstPartRequested, durationZeroGradientRequested, durationSecondPartRequested, discretizationSteps, forceSymmetry)
 
-startZeroGradientsIndex  = floor(durationFirstPartRequested/dt);
-startSecondPartIndex   = floor((durationFirstPartRequested+durationZeroGradientRequested)/dt);
-
-durationFirstPartActual = startZeroGradientsIndex * dt;
-durationSecondPartActual = (discretizationSteps-startSecondPartIndex)* dt;
-durationZeroGradientActual  = (startSecondPartIndex-startZeroGradientsIndex) * dt;
-totalTimeActual = durationFirstPartActual + durationSecondPartActual + durationZeroGradientActual;
-
-if durationZeroGradientActual > 0
-    zeroGradientAtIndex = (startZeroGradientsIndex:startSecondPartIndex);
+if ~forceSymmetry
```

Comment:
Style-wise, I think it's nicer to switch so that it's not a negation here.

Reply:
OK! You are the architect :)